client/state/utils.js

```diff
@@ -116,3 +116,54 @@ export function createReducer( initialState = null, customHandlers = {}, schema
 		return state;
 	};
 }
+
+/**
+ * Extends createReducer, returning a reducer function. The schema and return
+ * value of the enhanced reducer is augmented to create a nested object keyed
+ * by an assumed siteId property on each action handled by customHandlers.
+ *
+ * @see createReducer
+ *
+ * @param  {*}        initialSiteState Initial state per site ID
+ * @param  {Object}   customHandlers   Object mapping action types to state
+ *                                     action handlers
+ * @param  {?Object}  schema           JSON schema object for deserialization
+ *                                     validation
+ * @return {Function}                  Reducer function
+ */
+export function createSiteReducer( initialSiteState = null, customHandlers = {}, schema = null ) {
```

Comment:
We can leave this for another PR, but it'd be handy to explain what situations we use `createReducer` and `createSiteReducer` in [our-approach-to-data](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md). I just realized we use `createReducer` in our examples but never explain what it does.